### PR TITLE
to be reviewed

### PR DESCRIPTION
**Pull request by ejpres**
Sorry, hatte den Rechner in Essen gelassen über das lange Wochenende.

### CODE REVIEW

**Comment by splattael** [comment]
@ejpres Super, danke! Ich nehme mir morgen Vormittag Zeit für die Code-Review :nerd_face:

File: test/connect_four/board_test.rb [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        board.fields.each_with_index do |row, y|
+          row.each_with_index do |field, x|
+            field.must_equal '-' unless y == 7 && x == 3
+          end
+        end
+      end
+
+      it('stores a second stone in the same column on top of the first') do
+        board.put_stone('x', 3)
+        board.fields[6][3].must_equal 'x'
```
Comment (splattael) [inline test/connect_four/board_test.rb]: Could hiding the implementation detail could also improve the readability here?
```ruby
board.field_at(.., ..)
```

File: test/connect_four/game_test.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      end
+
+    end
+
+    describe '#start' do
+      describe 'with muted output and no winner' do
+        before do
+          mute
+          without_winner
+          def game.play symbol; @count ||= 0 ; @count += 1;end
```
Comment (splattael) [inline test/connect_four/game_test.rb]: Mh, everytime I have to monkeypatch an object I wonder if there is possibility to omit this kind of code? :grimacing: 
In this case I see 3 possibilities:
1. Inject an object into `Game` which provides a `play` method, so you can implement your own behaviour
2. Use a "mock" (or "stub"?)
3. Actually play the game (integration test)

Of course, monkeypatch is to explore things :see_no_evil: :hear_no_evil: :speak_no_evil:

File: test/test_helper.rb [hunk 3]
```diff
@@ -2,7 +2,36 @@
...
 
-require 'connect_four'
+require 'connect_four/game'
+require 'connect_four/board'
 
 class ConnectFourSpec < Minitest::Test
 end
+
+def initial_board_string
+  "1 2 3 4 5 6 7 8\n- - - - - - - -\n- - - - - - - -\n- - - - - - - -\n- - - - - - - -\n- - - - - - - -\n- - - - - - - -\n- - - - - - - -\n- - - - - - - -\n\n"
```
Comment (splattael) [inline test/test_helper.rb]: You can use Ruby's [squiggly heredoc](https://infinum.co/the-capsized-eight/multiline-strings-ruby-2-3-0-the-squiggly-heredoc):

```ruby
def initial_board_string
  <<~BOARD
    1 2 3 4 5 6 7 8
    - - - - - - - -
    - - - - - - - -
    - - - - - - - -
    - - - - - - - -
    - - - - - - - -
    - - - - - - - -
    - - - - - - - -
    - - - - - - - -

  BOARD
end
```

File: test/test_helper.rb [hunk 4]
```diff
@@ -2,7 +2,36 @@
...
+
+def board_and_prompt_string
+  "#{initial_board_string}player x >"
+end
+
+def setup_fields(array_of_coordinates, symbol, _board = board)
+  array_of_coordinates.each {|coordinates| _board.fields[coordinates.first][coordinates.last] = symbol}
+end
+
+def mute(obj = game)
```
Comment (splattael) [inline test/test_helper.rb]: This method goes away once "io" is injected in `Board.new`. See a comment above :grinning:

File: test/test_helper.rb [hunk 5]
```diff
@@ -2,7 +2,36 @@
...
 require 'minitest/autorun'
 
-require 'connect_four'
+require 'connect_four/game'
+require 'connect_four/board'
 
 class ConnectFourSpec < Minitest::Test
 end
+
+def initial_board_string
```
Comment (splattael) [inline test/test_helper.rb]: I would define all test helper methods inside `ConnectFourSpec`.
Currently, they are all global :fearful: :wink:

**Comment by 27sparks** [comment]
But be aware there's answers to come :-)
Been somewhat busy the last couple days, you might guess why...
(Or maybe I'll just wait until there's a while_with_index loop in ruby to have an as short replacement for my ternary operators and breaking inside them)

**Comment by splattael** [comment]
@27sparks / @ejpres Sure, no problem. I am happy to reopen this PR :+1: 
Keep on pushing :D

**Comment by 27sparks** [comment]
27sparks is my old(er) account... mobile phones never forget it seems, well, obviously.

**Pull request closed without merging**